File: imgazyobuzi/twitpic.go
package imgazyobuzi

import "regexp"

type twitpic resolverStruct

func (self *twitpic) ServiceName() (string, string) {
	return "twitpic", "Twitpic"
}

func (self *twitpic) Regex() *regexp.Regexp {
	if self.re == nil {
		self.re = regexp.MustCompile(`^https?://(?:www\.)?twitpic\.com/(?:show/\w+/)?(\w+)/?(?:\?.*)?(?:#.*)?$`)
	}
	return self.re
}

func (self *twitpic) Id(groups []string) string {
	return groups[1]
}

func (self *twitpic) Sizes(ctx *Context, groups []string) ([]ImageInfo, ResolvingErr) {
	return []ImageInfo{
		ImageInfo{
			"https://twitpic.com/show/large/" + groups[1],
			"https://twitpic.com/show/large/" + groups[1],
			"https://twitpic.com/show/thumb/" + groups[1],
			"",
		},
	}, ResolvingErr{}
}

var TwitpicInstance = new(twitpic)

File: ImgAzyobuziV3.Core/Resolvers/Twitpic.cs
using System.Collections.Generic;
using System.ComponentModel.Composition;
using System.Text.RegularExpressions;
using ImgAzyobuziV3.Core.DataModels;

namespace ImgAzyobuziV3.Core.Resolvers
{
    [Export(typeof(IResolver))]
    public class Twitpic : ResolverBase
    {
        public override string ServiceId
        {
            get { return "twitpic"; }
        }

        public override string ServiceName
        {
            get { return "Twitpic"; }
        }

        public override string PatternString
        {
            get { return @"^https?://(?:www\.)?twitpic\.com/(?:show/\w+/)?(\w+)/?(?:\?.*)?(?:#.*)?$"; }
        }

        public override string GetId(Match match)
        {
            return match.Groups[1].Value;
        }

        public override IReadOnlyCollection<ImageInfo> GetImages(ImgAzyobuziContext context, Match match)
        {
            var id = this.GetId(match);
            return new[]
            {
                new ImageInfo(
                    "https://twitpic.com/show/large/" + id,
                    "https://twitpic.com/show/large/" + id,
                    "https://twitpic.com/show/thumb/" + id
                )
            };
        }
    }
}
